backend/src/services/billing/billing.service.ts

```diff
@@ -0,0 +1,201 @@
+    rawPriceOfStudying,
+  }: BillingInitHoldStudentPaymentArgumentsDto): Promise<void> {
+    const menteeBalance = await this.#userService.getByIdMoneyBalance(menteeId);
+
+    const priceOfStudying =
+      rawPriceOfStudying * Billing.DEFAULT_STUDYING_PRICE_COEFFICIENT;
+
+    if (menteeBalance < priceOfStudying) {
+      throw new BillingError({
+        message: ExceptionMessage.NOT_ENOUGH_FUNDS_TO_PAY_FOR_MENTORS_SERVICES,
+      });
+    }
+
+    const newMenteeBalance = menteeBalance - priceOfStudying;
+
+    await this.#userDetailsService.updateMoneyBalance(
+      menteeId,
+      newMenteeBalance,
+    );
+
+    const transaction = await this.makeTransaction({
+      senderId: menteeId,
+      receiverId: mentorId,
+      amount: priceOfStudying,
+    });
+    await this.holdTransaction(transaction.id);
+  }
+
+  public getHoldTransactionBySenderAndReceiverId(
+    senderId: number,
+    receiverId: number,
+  ): Promise<TransactionGetAllItemResponseDto> {
+    return this.#transactionService.getHoldBySenderAndReceiverId(
+      senderId,
+      receiverId,
+    );
+  }
+
+  public initReplenish({
+    amount,
+    token,
+  }: StripeReplenishArgumentsDto): Promise<StripeApi.Charge> {
+    try {
+      return this.#stripe.charges.create({
+        source: token.id,
+        amount: amount * PaymentUnit.CENTS_IN_ONE_DOLLAR,
+        currency: Billing.BILLING_CURRENCY,
+      });
+    } catch (err) {
+      this.throwError(err);
+    }
+  }
+
+  public initWithdraw(
+    amount: number,
+  ): Promise<StripeApi.Response<StripeApi.Payout>> {
+    try {
+      return this.#stripe.payouts.create({
+        amount: amount * PaymentUnit.CENTS_IN_ONE_DOLLAR,
+        currency: Billing.BILLING_CURRENCY,
+      });
+    } catch (err) {
+      this.throwError(err);
+    }
+  }
+
+  public makeTransaction(
+    transactionCreateBody: TransactionCreateArgumentsDto,
+  ): Promise<TransactionGetAllItemResponseDto> {
+    return this.#transactionService.create(transactionCreateBody);
+  }
+
+  public holdTransaction(
+    transactionId: number,
+  ): Promise<TransactionGetAllItemResponseDto> {
+    return this.#transactionService.updateStatus({
+      transactionId,
+      newStatus: TransactionStatus.HOLD,
+    });
+  }
+
+  public fulfillTransaction(
+    transactionId: number,
+  ): Promise<TransactionGetAllItemResponseDto> {
+    return this.#transactionService.updateStatus({
+      transactionId,
+      newStatus: TransactionStatus.FULFILLED,
+    });
+  }
+
+  public rejectTransaction(
+    transactionId: number,
+  ): Promise<TransactionGetAllItemResponseDto> {
+    return this.#transactionService.updateStatus({
+      transactionId,
+      newStatus: TransactionStatus.REJECTED,
+    });
+  }
+
+  private throwError(err: unknown): never {
+    if (err instanceof StripeApi.errors.StripeError) {
+      throw new BillingError({
+        message: err.message,
+        status: err.statusCode,
+      });
+    }
+    throw new BillingError();
```

Comment:
what if there is an error message, but error not `instanceof StripeApi.errors.StripeError`?

Reply:
Then there will be thrown a default BillingError()